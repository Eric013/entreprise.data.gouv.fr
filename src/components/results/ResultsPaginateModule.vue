<template>
  <paginate
    ref="paginate"
    :container-class="'pagination'"
    :page-class = "'pagesButtons'"
    :prev-class="'pagesButtons'"
    :next-class="'pagesButtons'"
    :page-count="totalPageNumber"
    :initial-page="initialPage"
    :prev-text="'PrecÃ©dent'"
    :next-text="'Suivant'"
    :click-handler="selectPage">
  </paginate>
</template>

<script>
import Paginate from 'vuejs-paginate'
import Vue from 'vue'

Vue.component('paginate', Paginate)

export default {
  name: 'Results',
  props: ['totalPages'],
  data: function () {
    return {
      initialPage: parseInt(this.$store.state.route.query.page, 10) - 1,
      totalPageNumber: this.totalPages
    }
  },
  methods: {
    selectPage (pageNum) {
      if (this.$store.state.route.query.fullText) {
        this.$store.commit('setPage', pageNum)
        this.$store.dispatch('requestSearchFullText')
      } else {
        this.$store.dispatch('goToClearedHomePage')
      }
    }
  }
}
</script>

<style lang="scss">

.pagination {
	list-style: none;
  display: flex;
  justify-content: flex-end;
}

.pagesButtons {
  background: transparentize($color-light-blue, .8);
  border-radius: 3px;
  padding: 2px 5px;
  margin-left: 8px;

  &.disabled, &:first-child, &:last-child {
    background: none;
  }

  &.active {
    background-color: $color-light-blue;

    a {
      color: $color-white;
    }
  }
}

.pagesButtons a {
  color: $link-color;
  font-weight: 700;
}

</style>
