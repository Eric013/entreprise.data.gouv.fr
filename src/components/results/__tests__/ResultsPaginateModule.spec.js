import ResultsPaginateModule from '@/components/results/ResultsPaginateModule.vue'
import Vuex from 'vuex'
import { __createMocks as createStoreMocks } from '@/store/index.js'
import { createLocalVue, shallow, mount } from '@vue/test-utils'

const localVue = createLocalVue()
localVue.use(Vuex)
jest.mock('@/store/index.js')
jest.mock('@/components/results/ResultsPaginateModule.vue')

describe('ResultsPaginateModule.vue', () => {
  let storeMocks
  let wrapperRPM

  beforeEach(() => {
    storeMocks = createStoreMocks()

    wrapperRPM = shallow(ResultsPaginateModule, {
      localVue,
      store: storeMocks.store
    })
  })

  test('Data function returns the correct parsed int', () => {
  // Page is '5' in the store __mocks__, this should return 4
    expect(wrapperRPM.vm.initialPage).toBe(4)
  })

  test('Computed value totalPageNumber returns the right number', () => {
    expect(wrapperRPM.vm.totalPageNumber).toBe(10)
  })

  test('Method selectPage commit a new page number and request search if fullText present', () => {
    wrapperRPM.vm.selectPage(3)
    expect(storeMocks.mutations.setPage).toHaveBeenCalledWith(storeMocks.state, 3)
    expect(storeMocks.actions.requestSearch).toHaveBeenCalled()
  })

  test('Method selectPage activates goToClearedHomePage if fullText not present', () => {
    wrapperRPM.vm.$store.state.route.query.fullText = ''
    wrapperRPM.vm.selectPage(3)
    expect(storeMocks.actions.goToClearedHomePage).toHaveBeenCalled()
  })
})

describe('ResultsPaginateModule.vue Snapshot test', () => {
  const storeMocks = createStoreMocks()

  const wrapperRPM = mount(ResultsPaginateModule, {
      localVue,
      store: storeMocks.store
    })

  test('Snapshot test', () => {
    expect(wrapperRPM.vm.$el).toMatchSnapshot()
  })
}) 