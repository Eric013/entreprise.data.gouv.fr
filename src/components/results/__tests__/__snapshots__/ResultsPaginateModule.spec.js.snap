// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPaginateModule.vue Snapshot test Snapshot test 1`] = `
<ul
  class="pagination"
  data-v-82963a40=""
>
  <li
    class="pagesButtons"
    data-v-82963a40=""
  >
    <a
      data-v-82963a40=""
      tabindex="0"
    >
      Prec√©dent
    </a>
  </li>
   
  <li
    class="pagesButtons"
    data-v-82963a40=""
  >
    <a
      data-v-82963a40=""
      tabindex="0"
    >
      1
    </a>
  </li>
  <li
    class="pagesButtons disabled"
    data-v-82963a40=""
  >
    <a
      data-v-82963a40=""
      tabindex="0"
    >
      ...
    </a>
  </li>
  <li
    class="pagesButtons"
    data-v-82963a40=""
  >
    <a
      data-v-82963a40=""
      tabindex="0"
    >
      4
    </a>
  </li>
  <li
    class="pagesButtons active"
    data-v-82963a40=""
  >
    <a
      data-v-82963a40=""
      tabindex="0"
    >
      5
    </a>
  </li>
  <li
    class="pagesButtons"
    data-v-82963a40=""
  >
    <a
      data-v-82963a40=""
      tabindex="0"
    >
      6
    </a>
  </li>
  <li
    class="pagesButtons disabled"
    data-v-82963a40=""
  >
    <a
      data-v-82963a40=""
      tabindex="0"
    >
      ...
    </a>
  </li>
  <li
    class="pagesButtons"
    data-v-82963a40=""
  >
    <a
      data-v-82963a40=""
      tabindex="0"
    >
      10
    </a>
  </li>
   
  <li
    class="pagesButtons"
    data-v-82963a40=""
  >
    <a
      data-v-82963a40=""
      tabindex="0"
    >
      Suivant
    </a>
  </li>
</ul>
`;
